wgpu-hal/src/metal/adapter.rs

```diff
@@ -718,8 +718,12 @@ impl super::PrivateCapabilities {
             format_min_srgb_channels: if os_is_mac { 4 } else { 1 },
             format_b5: !os_is_mac,
             format_bc: os_is_mac,
-            format_eac_etc: !os_is_mac,
-            format_astc: Self::supports_any(device, ASTC_PIXEL_FORMAT_FEATURES),
+            format_eac_etc: !os_is_mac
+                // M1 in macOS supports EAC/ETC2
+                || device.supports_family(MTLGPUFamily::Apple7),
+            format_astc: Self::supports_any(device, ASTC_PIXEL_FORMAT_FEATURES)
+                // A14/M1 always support ASTC
+                || device.supports_family(MTLGPUFamily::Apple7),
```

Comment:
Shouldn't this be family 6? If ASTC is supported on ios family 2, then all gpu family 2+ should support this.

What family does the M1 report?

Reply:
> Shouldn't this be family 6? If ASTC is supported on ios family 2, then all gpu family 2+ should support this.

The `Self::supports_any(device, ASTC_PIXEL_FORMAT_FEATURES)` works for all `MTLGPUFamily` that have 'OS GPU families' defined in `MTLFeatureSet` but it misses those that don't. It looks like Apple6 doesn't have an 'OS GPU family' defined in `MTLFeatureSet` either. I think I saw that the feature sets are guaranteed to be supersets as the numbers get higher, but are the MTLGPUFamily necessarily supersets (i.e. Apple7 can only have features that are a superset of Apple6?) So this should either be:

- Apple7 is a superset of Apple6
```suggestion
                || device.supports_family(MTLGPUFamily::Apple6),
```
- Apple7 is not a superset of Apple6:
```suggestion
                || device.supports_family(MTLGPUFamily::Apple6)
                || device.supports_family(MTLGPUFamily::Apple7),
```

> What family does the M1 report?

M1 Max is Apple7. If someone has a 'plain' M1 or otherwise knows, it would be good to verify that all M1 are Apple7.

Reply:
> More recent GPUs have higher version numbers and support larger feature sets. A higher GPU version is always a superset of an earlier version in the same GPU family. For details on the individual families and version numbers, see [MTLGPUFamily](https://developer.apple.com/documentation/metal/mtlgpufamily).

So yeah, Apple6 is correct. Good catch!

Reply:
Fixed. Note that I left the ETC2 check as Apple7 because that explicit checks for not macOS or Apple7. Apple6 can't be used in macOS so it is covered by the first case.

Reply:
This should be resolved but I'll leave it unresolved so you all can see the discussion and disagree if you think I made a mistake.